subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -495,6 +482,35 @@ struct ztest_unit_test *ztest_get_next_test(const char *suite, struct ztest_unit
 	return NULL;
 }
 
+bool __weak ztest_get_list_test(void)
+{
+	return false;
+}
+
+int __weak ztest_list_tests(void)
+{
+	return 0;
+}
+
+bool __weak ztest_should_suite_run(const void *state, struct ztest_suite_node *suite)
+{
+	bool run_suite = true;
+
+	if (suite->predicate != NULL) {
+		run_suite = suite->predicate(state);
+	} else {
+		/* If predicate is NULL, only run this test once. */
+		run_suite = suite->stats.run_count == 0;
+	}
+
+	return run_suite;
+}
+
+bool __weak ztest_should_test_run(const char *suite, const char *test)
+{
+	return true;
+}
+
```

Comment:
Using `weak` references makes this vulnerable. Can we instead move them to a separate file `ztest_defaults.c` and conditionally include them in the sources via CMakeLists like we did with `ztest_posix.c`? Basically:

```
if ARCH_POSIX
  add ztest_posix.c
else
  add ztest_defaults.c
endif
```

This would allow other architectures to possibly override these as well without using the weak reference.

Reply:
Done